src/core/coap/coap.cpp

```diff
@@ -1020,9 +1020,11 @@ Coap::Coap(Instance &aInstance)
 
 otError Coap::Start(uint16_t aPort, otNetifIdentifier aNetifIdentifier)
 {
-    otError error;
+    otError error        = OT_ERROR_NONE;
     bool    socketOpened = false;
 
+    VerifyOrExit(!mSocket.IsBound(), OT_NOOP);
```

Comment:
According to the available return values in the header file,  `OT_ERROR_ALREADY` might be more appropriate here?

https://github.com/openthread/openthread/blob/25ae9b073955a4fa00d8b0b2ed368c1ac77fec8f/src/core/coap/coap.hpp#L607-L616

Reply:
Agree that it's better for this PR to return `OT_ERROR_ALREADY`. 

But I think in general we should avoid using `OT_ERROR_ALREADY`. We can refactor it in a separate PR.